nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -0,0 +1,1076 @@
+                            try {
+                                ps.setQueryTimeout(queryTimeout); // timeout in seconds
+                            } catch (SQLException se) {
+                                // If the driver doesn't support query timeout, then assume it is "infinite". Allow a timeout of zero only
+                                if (queryTimeout > 0) {
+                                    throw se;
+                                }
+                            }
+
+                            Record currentRecord;
+                            List<Integer> fieldIndexes = sqlHolder.getFieldIndexes();
+
+                            while ((currentRecord = recordParser.nextRecord()) != null) {
+                                Object[] values = currentRecord.getValues();
+                                if (values != null) {
+                                    if (fieldIndexes != null) {
+                                        for (int i = 0; i < fieldIndexes.size(); i++) {
+                                            ps.setObject(i + 1, values[fieldIndexes.get(i)]);
+                                        }
+                                    } else {
+                                        // If there's no index map, assume all values are included and set them in order
+                                        for (int i = 0; i < values.length; i++) {
+                                            ps.setObject(i + 1, values[i]);
+                                        }
+                                    }
+                                    ps.addBatch();
+                                }
+                            }
+
+                            log.debug("Executing query {}", new Object[]{sqlHolder});
+                            ps.executeBatch();
+                            session.transfer(flowFile, REL_SUCCESS);
+                            session.getProvenanceReporter().send(flowFile, jdbcURL);
+
+                        } catch (final SQLNonTransientException e) {
+                            log.error("Failed to update database for {} due to {}; routing to failure", new Object[]{flowFile, e});
+                            flowFile = session.penalize(flowFile);
+                            session.transfer(flowFile, REL_FAILURE);
+                        } catch (final SQLException e) {
+                            log.error("Failed to update database for {} due to {}; it is possible that retrying the operation will succeed, so routing to retry",
+                                    new Object[]{flowFile, e});
+                            flowFile = session.penalize(flowFile);
+                            session.transfer(flowFile, REL_RETRY);
+                        }
+                    }
+                } catch (final MalformedRecordException | SchemaNotFoundException | IOException e) {
+                    throw new ProcessException("Failed to determine schema of data records for " + flowFile, e);
```

Comment:
Confirmed that a FlowFile is routed to 'failure' when a schema is not found with specified name. Thanks!